pkg/database/database.go

```diff
@@ -7,17 +7,24 @@ import (
 	"context"
 	"database/sql"
 	"fmt"
+
 	"github.com/kubearchive/kubearchive/pkg/models"
 	_ "github.com/lib/pq"
 )
 
 const (
-	resourceTableName = "resource"
-	resourcesQuery    = "SELECT data FROM %s WHERE kind=$1 AND api_version=$2"
+	resourceTableName   = "resource"
+	resourcesQuery      = "SELECT data FROM %s WHERE kind=$1 AND api_version=$2"
+	resourceIdQuery     = `SELECT id FROM %s WHERE kind=$1 AND api_version=$2 AND cluster=$3 AND cluster_uid=$4 AND name=$5 AND namespace=$6`
```

Comment:
If you need this query (and I don't think you do, see my earlier comment) you should just be able to query by the resource UUID, which is guaranteed to be unique.  

In fact,  UUID should be the primary key on the table.

Reply:
I've added using UUID as the primary key as a project discussion topic

Reply:
uuid is now the primary key and upsert is done using uuid